packages/opentelemetry-tracing/src/export/ReadableSpan.ts

```diff
@@ -39,6 +39,6 @@ export interface ReadableSpan {
   readonly events: TimedEvent[];
   readonly duration: HrTime;
   readonly ended: boolean;
-  readonly resource: Resource;
+  readonly resource: Resource | Promise<Resource>;
```

Comment:
Does this need to be a Promise? Since we know we will await it in the tracer, I see no reason to complicate the types of the export pipeline

Reply:
I couldn't find a better way so that it has to be like this or I can remove `Resource | Promise<Resource>;` from almost everywhere but I will have to cast the resource to unknown and then to Resource. This might be a bit hacky as I will have to fake the typescript but that's the only way that works. So either we have `Resource | Promise<Resource>;` everywhere or we fake ts and cast it to "Resource" in tests.

Reply:
fixed